config/variable_binning.py

```diff
@@ -5,24 +5,38 @@
                           'angle_bl' : [round( i * 0.2, 2 ) for i in range ( int( 4 / 0.2 ) + 1 )],
                           'Mjj' : [i * 25 for i in range ( int( 500 / 25 ) + 1 )],
                           }
+
 bin_edges = {
-'MET' : [0.0, 42.0, 100.0, 332.0],
-'WPT' : [0.0, 44.0, 87.0, 134.0, 197.0, 449.0],
-'NJets' : [3.5, 4.5, 5.5, 6.5, 10.5],
-'HT' : [100.0, 204.0, 248.0, 298.0, 354.0, 418.0, 488.0, 561.0, 642.0, 755.0, 1207.0],
-'ST' : [123.0, 298.0, 359.0, 425.0, 501.0, 587.0, 682.0, 790.0, 898.0, 1330.0],
-'lepton_pt' : [23.0, 31.0, 37.0, 42.0, 48.0, 54.0, 61.0, 69.0, 79.0, 92.0, 113.0, 197.0],
-'abs_lepton_eta' : [0.0, 0.117, 0.234, 0.357, 0.483, 0.609, 0.741, 0.885, 1.041, 1.218, 1.416, 2.208],
+# 'MET' : [0.0, 42.0, 99.0, 327.0],
+# 'WPT' : [0.0, 44.0, 87.0, 134.0, 194.0, 270.0, 574.0],
+# 'NJets' : [3.5, 4.5, 5.5, 6.5, 10.5],
+# 'HT' : [100.0, 193.0, 240.0, 293.0, 350.0, 412.0, 481.0, 553.0, 630.0, 723.0, 1095.0],
+# 'ST' : [123.0, 299.0, 362.0, 432.0, 510.0, 596.0, 690.0, 795.0, 907.0, 1355.0],
+# 'lepton_pt' : [23.0, 31.0, 36.0, 41.0, 46.0, 51.0, 56.0, 62.0, 69.0, 77.0, 87.0, 101.0, 124.0, 216.0],
+# 'abs_lepton_eta' : [0.0, 0.099, 0.198, 0.303, 0.402, 0.504, 0.609, 0.717, 0.834, 0.960, 1.095, 1.245, 1.416, 1.665, 2.661],}
+'WPT' : [0.0, 44.0, 87.0, 134.0, 194.0, 270.0, 366.0, 750.0],
+'MET' : [0.0, 42.0, 99.0, 257.0, 667.0],
+'NJets' : [3.5, 4.5, 5.5, 6.5, 8.5, 16.5],
+'HT' : [100.0, 193.0, 240.0, 293.0, 350.0, 412.0, 481.0, 553.0, 630.0, 718.0, 808.0, 912.0, 1015.0, 1148.0, 1680.0],
+'ST' : [123.0, 299.0, 362.0, 432.0, 510.0, 596.0, 690.0, 795.0, 907.0, 1026.0, 1150.0, 1291.0, 1855.0],
+'lepton_pt' : [23.0, 25.0, 27.0, 29.0, 31.0, 33.0, 35.0, 37.0, 39.0, 41.0, 43.0, 45.0, 47.0, 49.0, 51.0, 53.0, 55.0, 57.0, 59.0, 61.0, 63.0, 66.0, 69.0, 72.0, 75.0, 78.0, 81.0, 84.0, 87.0, 90.0, 94.0, 98.0, 102.0, 106.0, 110.0, 114.0, 119.0, 125.0, 131.0, 139.0, 149.0, 163.0, 184.0, 268.0],
+'abs_lepton_eta' : [0.0, 0.018, 0.036000000000000004, 0.054000000000000006, 0.07200000000000001, 0.09000000000000001, 0.108, 0.126, 0.14400000000000002, 0.162, 0.18, 0.198, 0.216, 0.234, 0.252, 0.27, 0.28800000000000003, 0.306, 0.324, 0.342, 0.36, 0.378, 0.396, 0.41400000000000003, 0.432, 0.45, 0.468, 0.486, 0.504, 0.522, 0.54, 0.558, 0.5760000000000001, 0.594, 0.615, 0.633, 0.654, 0.675, 0.6960000000000001, 0.714, 0.735, 0.756, 0.777, 0.798, 0.8190000000000001, 0.84, 0.861, 0.882, 0.906, 0.927, 0.9510000000000001, 0.975, 0.999, 1.023, 1.047, 1.071, 1.095, 1.1219999999999999, 1.149, 1.176, 1.2029999999999998, 1.23, 1.257, 1.2839999999999998, 1.3139999999999998, 1.3439999999999999, 1.3739999999999999, 1.404, 1.4369999999999998, 1.4789999999999999, 1.53, 1.5779999999999998, 1.617, 1.656, 1.698, 1.74, 1.785, 1.8359999999999999, 1.884, 1.938, 1.9949999999999999, 2.223],
```

Comment:
Personally I would edit the bins by hand to smooth them out.
Judging by the first few bins, the size seems to be 0.018